kubeflow/common/basic-auth.libsonnet

```diff
@@ -0,0 +1,175 @@
+        annotations: {
+          "getambassador.io/config":
+            std.join("\n", [
+              "---",
+              "apiVersion: ambassador/v0",
+              "kind:  AuthService",
+              "name: " + params.name,
+              "auth_service: " + params.name + "." + params.namespace + ":8085",
+              'allowed_headers:\n- "x-from-login"',
+            ]),
+        },  //annotations
+      },
+      spec: {
+        ports: [
+          {
+            port: 8085,
+            targetPort: 8085,
+          },
+        ],
+        selector: {
+          app: params.name,
+        },
+        type: "ClusterIP",
+      },
+    },
+    authService:: authService,
+
+    local authDeployment = {
+      apiVersion: "extensions/v1beta1",
+      kind: "Deployment",
+      metadata: {
+        name: params.name,
+        namespace: params.namespace,
+
+      },
+      spec: {
+        replicas: 1,
+        strategy: {
+          type: "RollingUpdate",
+        },
+        template: {
+          metadata: {
+            labels: {
+              app: params.name,
+            },
+          },
+          spec: {
+            containers: [
+              {
+                image: params.image,
+                name: "app",
+                workingDir: "/opt/kubeflow",
+                command: [
+                  "/opt/kubeflow/gatekeeper",
+                ],
+                args: [
+                  "--username=" + params.username,
+                  "--pwhash=" + params.pwhash,
+                ],
+                ports: [
+                  {
+                    containerPort: 8085,
+                  },
+                ],
+              },
+            ],
+          },
+        },
+      },
+    },
+    authDeployment:: authDeployment,
+
+    local loginService = {
+      apiVersion: "v1",
+      kind: "Service",
+      metadata: {
+        labels: {
+          app: ui_name,
+        },
+        name: ui_name,
+        namespace: params.namespace,
+        annotations: {
+          "getambassador.io/config":
+            std.join("\n", [
+              "---",
+              "apiVersion: ambassador/v0",
+              "kind:  Mapping",
+              "name: kflogin-mapping",
+              "prefix: /kflogin",
+              "rewrite: /kflogin",
+              "timeout_ms: 300000",
+              "service: " + ui_name + "." + params.namespace,
+              "use_websocket: true",
+            ]),
+        },  //annotations
+      },
+      spec: {
+        ports: [
+          {
+            port: 80,
+            targetPort: 5000,
+          },
+        ],
+        selector: {
+          app: ui_name,
+        },
+        type: "ClusterIP",
+      },
+    },
+    loginService:: loginService,
+
+    local loginDeployment = {
+      apiVersion: "extensions/v1beta1",
+      kind: "Deployment",
+      metadata: {
+        name: ui_name,
+        namespace: params.namespace,
+
+      },
+      spec: {
+        replicas: 1,
```

Comment:
Add a comment here explaining this should always be 1 because the cookie is stored in memory so if we use more than 1 replica we will have problems because not all replicas will have the cookie.

Reply:
Comments updated